translator/translator/translator.py

```diff
@@ -0,0 +1,216 @@
+            return
+        # Insert the message at the top of the list.
+        self.messages.insert(
+            0,
+            Message(
+                original_text=text,
+                text=self.output,
+                created_at=datetime.now().strftime("%B %d, %Y %I:%M %p"),
+                to_lang=self.lang,
+            ),
+        )
+        # Manually reset the form input, since input in this app is fully controlled, it does not reset on submit.
+        self.text = ""
+
+
+# Define views.
+
+
+def header() -> rx.Component:
+    """The header and the description."""
+    return rx.box(
+        rx.heading("Translator 🗺", size="8"),
+        rx.text(
+            "Translate things and post them as messages!",
+            color_scheme="gray",
+        ),
+        margin_bottom="1rem",
+    )
+
+
+def down_arrow() -> rx.Component:
+    return rx.vstack(
+        rx.icon(
+            tag="arrow_down",
+            color="gray",
+        ),
+    )
+
+
+def text_box(text) -> rx.Component:
+    return rx.text(
+        text,
+        background_color="white",
+        padding="1rem",
+        border_radius="8px",
+    )
+
+
+def past_translation(message: Message) -> rx.Component:
+    """A layout that contains a past translation."""
+    return rx.box(
+        rx.flex(
+            text_box(message.original_text),
+            down_arrow(),
+            text_box(message.text),
+            rx.hstack(
+                rx.text(message.to_lang),
+                rx.text("·"),
+                rx.text(message.created_at),
+                spacing="2",
+                font_size="0.8rem",
+                color="var(--gray-11)",
+            ),
+            spacing="2",
+            direction="column",
+        ),
+        background_color="var(--gray-3)",
+        padding="1rem",
+        border_radius="8px",
+    )
+
+
+def smallcaps(text: str, **props):
+    """A smallcaps text component."""
+    return rx.text(
+        text,
+        font_size="0.7rem",
+        font_weight="bold",
+        text_transform="uppercase",
+        letter_spacing="0.05rem",
+        **props,
+    )
+
+
+def output():
+    """The output layout that contains the translated text."""
+    return rx.stack(
+        smallcaps(
+            "Output",
+            color="var(--gray-9)",
+            background_color="white",
+            position="absolute",
+            top="-0.5rem",
+        ),
+        rx.text(State.output, size="4"),
+        padding="1rem",
+        border="1px solid var(--gray-7)",
+        border_radius="8px",
+        position="relative",
+    )
+
+
+def translation_form() -> rx.Component:
+    """This is a form that contains:
+    - Text input: what to translate
+    - Select: the destination language
+    - Output: the translation
+    - Submit button: to post the text input as part of the past translations view
+    """
+    return rx.form(
+        rx.flex(
+            rx.input.input(
```

Comment:
```suggestion
            rx.input(
```

rx.input is just an alias for the textfield input now, no need for the double input